### PR TITLE
feat(web): mailersend integration

### PR DESCRIPTION
**Pull request by LetItRock**
### What change does this PR introduce? 

New email integration for the MailerSend service.
The provider was already created some time ago.
I've also verified this integration and an email sending functionality.

### Why was this change needed?

This integration allows our clients to use the MailerSend service as their mailing platform.

### Other information (Screenshots)
Screenshot 2022-11-26 at 23 52 30

Screenshot 2022-11-26 at 23 52 38

Screenshot 2022-11-26 at 23 52 58

### CODE REVIEW

File: apps/api/src/app/events/services/mail-service/handlers/mailersend.handler.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { ChannelTypeEnum } from '@novu/shared';
+import { ICredentials } from '@novu/dal';
+import { MailersendEmailProvider } from '@novu/mailersend';
+
+import { BaseHandler } from './base.handler';
+
+export class MailerSendHandler extends BaseHandler {
```
Comment (LetItRock) [inline apps/api/src/app/events/services/mail-service/handlers/mailersend.handler.ts:7]: the MailerSend handler that is used by MailFactory

File: apps/api/src/app/events/services/mail-service/handlers/mailersend.handler.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  constructor() {
+    super('mailersend', ChannelTypeEnum.EMAIL);
+  }
+
+  buildProvider(credentials: ICredentials, from?: string) {
+    this.provider = new MailersendEmailProvider({
+      apiKey: credentials.apiKey,
```
Comment (LetItRock) [inline apps/api/src/app/events/services/mail-service/handlers/mailersend.handler.ts:14]: the only API key (token) is required to use their client library

File: libs/shared/src/consts/providers/channels/email.ts [hunk 3]
```diff
@@ -95,4 +96,12 @@ export const emailProviders: IProviderConfig[] = [
...
     logoFileName: { light: 'netcore.png', dark: 'netcore.png' },
   },
+  {
+    id: EmailProviderIdEnum.MailerSend,
+    displayName: 'MailerSend',
+    channel: ChannelTypeEnum.EMAIL,
+    credentials: mailerSendConfig,
+    docReference: 'https://developers.mailersend.com/',
+    logoFileName: { light: 'mailersend.svg', dark: 'mailersend.svg' },
+  },
```
Comment (LetItRock) [inline libs/shared/src/consts/providers/channels/email.ts:106]: the email provider that is used by the Integration page

File: libs/shared/src/consts/providers/credentials/provider-credentials.ts [hunk 4]
```diff
@@ -228,6 +228,16 @@ export const sesConfig: IConfigCredentials[] = [
...
 
+export const mailerSendConfig: IConfigCredentials[] = [
+  {
+    key: CredentialsKeyEnum.ApiKey,
+    displayName: 'API Key',
+    type: 'string',
+    required: true,
+  },
+  ...mailConfigBase,
+];
```
Comment (LetItRock) [inline libs/shared/src/consts/providers/credentials/provider-credentials.ts:239]: the config for the integration modal

File: providers/mailersend/src/lib/mailersend.provider.ts [hunk 5]
```diff
@@ -42,7 +44,8 @@ export class MailersendEmailProvider implements IEmailProvider {
     const attachments = this.getAttachments(options.attachments);
 
     const emailParams = new EmailParams()
-      .setFrom(options.from)
+      .setFrom(options.from ?? this.config.from)
+      .setFromName(this.config.senderName)
```
Comment (LetItRock) [inline providers/mailersend/src/lib/mailersend.provider.ts:48]: the provider was already created I've just adjusted it with these two fields

**Review by scopsy — APPROVED** [review]
Looks good!

**Pull request merged**